xl-v1.go

```diff
@@ -144,21 +144,27 @@ func newXLObjects(disks, ignoredDisks []string) (ObjectLayer, error) {
 	}
 
 	// Handles different cases properly.
-	switch reduceFormatErrs(sErrs, len(storageDisks)) {
-	case errUnformattedDisk:
-		// All drives online but fresh, initialize format.
-		if err := initFormatXL(storageDisks); err != nil {
-			return nil, fmt.Errorf("Unable to initialize format, %s", err)
+	for _, err := range reduceFormatErrs(sErrs, len(storageDisks)) {
+		switch err {
```

Comment:
Well, I thought that I need to process multiple errors when some disks are corrupted and some are fresh. But I see now this is a mistake. However, healFormatXLCorruptedDisks() will need to init unformatted disks too.. I really wanted to separate healing process from adding fresh disks but the information distributed over all the disks are fully connected.

So I see why you are considered the state of all disks rather than the state of each one of the disks. Will fix that